### PR TITLE
feat: optionally enable/disable gateway api

### PR DESCRIPTION
**Pull request by coltmcnealy-lh**

### CODE REVIEW

**Review by yesidevelop — APPROVED** [review]

File: charts/lh-operator/templates/role.yaml [hunk 1]
```diff
@@ -22,9 +22,11 @@ rules:
 - apiGroups: ["littlehorse.io"]
   resources: ["*"]
   verbs: ["*"]
+{{- if .Values.strimzi.enabled }}
```
Comment (mijailr) [inline charts/lh-operator/templates/role.yaml]:
```suggestion
{{- if .Values.gatewayApi.enabled }}
```

**Review by mijailr — COMMENTED** [review]
Apply the suggestion

File: charts/lh-operator/templates/role.yaml (hunk 1 above)
Comment (MateoRojas) [reply charts/lh-operator/templates/role.yaml]: Great catch! Let me update it right now :)

**Review by MateoRojas — APPROVED** [review]

File: charts/lh-operator/templates/role.yaml (hunk 1 above)
Comment (coltmcnealy-lh) [reply charts/lh-operator/templates/role.yaml]: Wait a minute how did I miss that? I tested it. Thanks Mijaíl!

Comment (coltmcnealy-lh) [reply charts/lh-operator/templates/role.yaml]: Oh, I see. This is the `Role` not the `Deployment`, which is why my tests worked as expected. I didn't test the case when `strimzi.enabled=false` and `gatewayApi.enabled=true`. That one would have failed.

Thansk @MateoRojas and @mijailr

**Pull request merged**